assignment7.txt

```diff
@@ -0,0 +1,25 @@
+1.
+The difference between ++*P, *p++ and *++p is that:
+++*P increases the value of *p by one
+*p++ moves up the location of the pointer by one
+*++p will do exactly what ++*p does
+
+2. 
+Yes, the order is guaranteed
+
+3.
+For fuction parameters
+for dynamic programming
+Pointer enables you to do things that are otherwise impossible, it also speeds up the operation
+
+4.
+4.1 invalid because b and c are undeclared.
+4.2 invalid because of undeclared variables.
+4.3 I am not sure what this means.
+4.4 10 because the pointer of a string prints the first number.
+4.5 10 because a[0] = 10.
+4.6 12 because a = 10 and *p = a + 2.
```

Comment:
hm, not quite! Since a is a pointer to the first element of the array, the increment by 2 would move the pointer two spots over to 12. so 12 is the answer.